### PR TITLE
WELD-2399 Add WeldContainer.current method.

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

File: environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java [hunk 1]
```diff
@@ -122,6 +122,17 @@ public static WeldContainer instance(String id) {
         }
     }
 
+    public static WeldContainer current() {
```
Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: Where's the contract? ;-)

Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: I mean javadoc - we should support autocompletion-based development...

Comment (manovotn) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: Yea I know what you mean, I was too hungry and pushed it without docs :-P

File: environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java [hunk 2]
```diff
@@ -123,6 +123,23 @@ public static WeldContainer instance(String id) {
     }
 
     /**
+     * A convenient method for CDI.current(). Returns current WeldContainer instance
+     * if there is exactly one instance running. Throws IllegalStateException in any other case.
+     * @return current WeldContainer instance iff exactly one instance exists
```
Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: `iff`typo and pls use `{@link }` for classes and methods.

Comment (manovotn) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: `iff` is not a typo, it's an abbreviation for `if and only if`

Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: Well, if it's abbreviation then use upper case please. But I would rather avoid using similar abbreviations in javadoc...

File: environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java [hunk 3]
```diff
@@ -123,6 +123,23 @@ public static WeldContainer instance(String id) {
     }
 
     /**
+     * A convenient method for {@link CDI.current()}. Returns current {@link WeldContainer} instance
+     * if there is exactly one instance running. Throws {@link IllegalStateException} in any other case.
+     * @return Current {@link WeldContainer} instance if and only if exactly one instance exists
+     * @throws {@link IllegalStateException} if there is 0 or more than 1 instance running
+     */
+    public static WeldContainer current() {
+        List<String> ids = WeldContainer.getRunningContainerIds();
```
Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: Just a minor impl note - every time you call `getRunningContainerIds()` you get a new immutable copy of `WeldContainer.RUNNING_CONTAINER_IDS`. So you could probably use the list directly.

Comment (manovotn) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: I know, I used that as I think it's a safe way to handle lists.
But you are right that here we can simply use the original list; I will change the code.

Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java]: You're right it's more safe to use the copy here.

**Pull request merged**